source/common/common/fine_grain_logger.cc

```diff
@@ -78,7 +81,8 @@ std::string FineGrainLogContext::listFineGrainLoggers() ABSL_LOCKS_EXCLUDED(fine
   std::string info = "";
   absl::ReaderMutexLock l(&fine_grain_log_lock_);
   for (const auto& it : *fine_grain_log_map_) {
-    info += fmt::format("   {}: {}\n", it.first, static_cast<int>(it.second->level()));
+    absl::StrAppend(&info, "   ",
+                    absl::StrJoin(std::make_tuple(it.first, it.second->level()), ": "), "\n");
```

Comment:
was thinking more along the lines of:

```
// Joins a `std::map`, with each key-value pair separated by an equals
// sign. This pattern would also work with, say, a
// `std::vector<std::pair<>>`.
std::map<std::string, int> m = {{"a", 1}, {"b", 2}, {"c", 3}};
std::string s = absl::StrJoin(m, "\n", absl::PairFormatter(" : "));
// Produces the string 
// a : 1
// b : 2
// c : 3
```

Reply:
sound good. The underlying pair is spdlog pointer which cannot be directly formatted, but we can add a lambda to format it.